c10/xpu/test/impl/XPUCachingAllocatorTest.cpp

```diff
@@ -0,0 +1,49 @@
+#include <gtest/gtest.h>
+
+#include <c10/xpu/XPUCachingAllocator.h>
+
+#define ASSERT_EQ_XPU(X, Y) \
+  {                         \
+    bool _isEQ = X == Y;    \
+    ASSERT_TRUE(_isEQ);     \
+  }
+
+bool has_xpu() {
+  return c10::xpu::device_count() > 0;
+}
+
+TEST(XPUCachingAllocatorTest, GetXPUAllocator) {
+  auto* allocator = c10::xpu::XPUCachingAllocator::get();
+
+  auto _500mb = 500 * 1024 * 1024;
+  auto buffer = allocator->allocate(_500mb);
```

Comment:
should you also check the allocated buffer here?

Reply:
done.